libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -790,22 +784,25 @@ static DNNReturnType execute_model_tf(const DNNModel *model, const char *input_n
         return DNN_ERROR;
     }
 
+    output = av_malloc_array(nb_output, sizeof(*output));
     for (uint32_t i = 0; i < nb_output; ++i) {
-        output.height = TF_Dim(output_tensors[i], 1);
-        output.width = TF_Dim(output_tensors[i], 2);
-        output.channels = TF_Dim(output_tensors[i], 3);
-        output.data = TF_TensorData(output_tensors[i]);
-        output.dt = TF_TensorType(output_tensors[i]);
-
+        output[i].height = TF_Dim(output_tensors[i], 1);
+        output[i].width = TF_Dim(output_tensors[i], 2);
+        output[i].channels = TF_Dim(output_tensors[i], 3);
+        output[i].data = TF_TensorData(output_tensors[i]);
+        output[i].dt = TF_TensorType(output_tensors[i]);
+    }
+    {
+        //it only support 1 output if it's frame in & frame out
         if (do_ioproc) {
             if (tf_model->model->frame_post_proc != NULL) {
-                tf_model->model->frame_post_proc(out_frame, &output, tf_model->model->filter_ctx);
+                tf_model->model->frame_post_proc(out_frame, output, tf_model->model->filter_ctx);
             } else {
-                ff_proc_from_dnn_to_frame(out_frame, &output, ctx);
+                ff_proc_from_dnn_to_frame(out_frame, &output[0], ctx);
```

Comment:
callback for detection missed

Reply:
Yes, this patch does not work well, detect callback func is in the last patch. Maybe current patch set is not resonable, i will think again.

Reply:
yes, thanks.